reset/reset.go

```diff
@@ -0,0 +1,89 @@
+package reset
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+type NodeService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PersistentVolumeService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PersistentVolumeClaimService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type StorageClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
+
+type PriorityClassService interface {
+	DeleteCollection(ctx context.Context, lopts metav1.ListOptions) error
+}
```

Comment:
They all have the same interface. So, we can use just one interface for better abstraction. 
That way, when new resource is supported in the future, the changes to the Reset service will be small.